Content.Server/GameObjects/Components/Buckle/BuckleComponent.cs

```diff
@@ -275,6 +276,8 @@ public override bool TryBuckle(IEntity user, IEntity to)
             AppearanceComponent?.SetData(BuckleVisuals.Buckled, true);
 
             BuckledTo = strap;
+            EntityBuckledTo = BuckledTo.Owner.Uid;
```

Comment:
Make EntityBuckledTo virtual or abstract and then override it in this server component to  return the uid of buckledTo, like `public override EntityUid? EntityBuckledTo => _buckledTo?.Owner.Uid`, then you only need to keep track of one property.

Reply:
If I make this change the fix stops working, I should rename the variable to LastEntityBuckledTo then, because if the variable changes before the player gets out of the collider it gets pushed out of the collider.

Reply:
Ah, my bad then. Renaming it to `LastEntityBuckledTo` would be good yes.

Reply:
Nah my bad for choosing a bad name for the variable :)